homeassistant/components/lunatone/light.py

```diff
@@ -0,0 +1,108 @@
+"""Platform for Lunatone light integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.light import ColorMode, LightEntity
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import LunatoneConfigEntry, LunatoneDevicesDataUpdateCoordinator
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: LunatoneConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Lunatone Light platform."""
+    coordinator_info = config_entry.runtime_data.coordinator_info
+    coordinator_devices = config_entry.runtime_data.coordinator_devices
+
+    info_api = coordinator_info.info_api
+    devices_api = coordinator_devices.devices_api
+
+    async_add_entities(
+        [
+            LunatoneLight(coordinator_devices, device.id, info_api.serial_number)
+            for device in devices_api.devices
+        ]
+    )
+
+
+class LunatoneLight(
+    CoordinatorEntity[LunatoneDevicesDataUpdateCoordinator], LightEntity
+):
+    """Representation of a Lunatone light."""
+
+    _attr_color_mode = ColorMode.ONOFF
+    _attr_supported_color_modes = {ColorMode.ONOFF}
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        coordinator: LunatoneDevicesDataUpdateCoordinator,
+        device_id: int,
+        interface_serial_number: int,
+    ) -> None:
+        """Initialize a LunatoneLight."""
+        super().__init__(coordinator=coordinator)
+        self._device_id = device_id
+        self._interface_serial_number = interface_serial_number
+        self._device = self.coordinator.device_api_mapping.get(self._device_id)
+        self._attr_unique_id = f"{interface_serial_number}-device{device_id}"
+        self._apply_device_state()
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        name = self._device.name if self._device is not None else None
```

Comment:
```suggestion
        assert self.unique_id
        name = self._device.name if self._device is not None else None
```
I think we can just assert unique_id, since it is set in init, and remove the `str()` cast